src/aap_eda/services/activation/manager.py

```diff
@@ -0,0 +1,1089 @@
+                f"Monitor operation: activation id: {self.db_instance.id} "
+                f"Container is in an error state. {container_status.message}"
+            )
+            LOGGER.error(msg)
+            self._cleanup()
+            self._error_instance(msg)
+            self._error_activation(msg)
+            raise exceptions.ActivationMonitorError(msg)
+
+        # we don't expect stopping status for the container
+        if container_status.status == ActivationStatus.STOPPED:
+            raise exceptions.ActivationMonitorError(
+                f"Container {self.latest_instance.activation_pod_id} "
+                "is in an stopped state.",
+            )
+
+    def update_logs(self):
+        """Update the logs of the latest instance of the activation."""
+        log_handler = self.container_logger_class(self.latest_instance.id)
+        try:
+            self._check_latest_instance_and_pod_id()
+        except (
+            exceptions.ActivationInstanceNotFound,
+            exceptions.ActivationInstancePodIdNotFound,
+        ):
+            msg = (
+                f"Update logs operation failed for activation id: "
+                f"{self.db_instance.id} No instance or pod id found."
+            )
+            LOGGER.error(msg)
+            # Alex: Not sure if we want to change the status here.
+            # For now, we are not changing the status of the activation
+            return
+
+        try:
+            self.container_engine.update_logs(
+                container_id=self.latest_instance.activation_pod_id,
+                log_handler=log_handler,
+            )
+
+        # We don't have identified cases where we want to stop the workflow
+        # because of a log error.
+        # We are just logging the error and continue.
+        except engine_exceptions.ContainerEngineError as exc:
+            msg = (
+                f"Logs for activation {self.db_instance.id} could not be "
+                f"fetch. Reason: {exc}"
+            )
+            LOGGER.error(msg)
+            log_handler.write(msg, flush=True)
+            # Alex: Not sure if we want to change the status here.
+            # For now, we are not changing the status of the activation
+            return
+
+    def _create_activation_instance(self):
+        try:
+            models.ActivationInstance.objects.create(
+                activation=self.db_instance,
+                name=self.db_instance.name,
+                status=ActivationStatus.STARTING,
+                git_hash=self.db_instance.git_hash,
+            )
+        except IntegrityError as exc:
+            msg = (
+                f"Activation {self.db_instance.id} failed to create "
+                f"activation instance. Reason: {exc}"
+            )
+            self._error_activation(msg)
+            raise exceptions.ActivationStartError(msg) from exc
+
+    def _build_container_request(self) -> ContainerRequest:
+        if self.db_instance.extra_var:
```

Comment:
@Dostonbek1 this is the check we are doing at runtime for the extra_vars once its checked during create we will remove this one